js/editor-ta.js

```diff
@@ -0,0 +1,304 @@
+  this.mirror_.innerText = this.textarea_.value;
+
+  // If we have a blank line at the end we have to add a character to get the
+  // height to set correctly
+  if (this.textarea_.value[this.textarea_.value.length-1] === '\n') {
+    this.mirror_.innerText += 'A';
+  }
+  this.updateLineNumbers();
+  this.updateTextArea();
+  // TODO(zafzal): need to emit a docChange event to each tab can know a change occured...
+};
+
+EditorTextArea.prototype.updateTextArea = function() {
+  const calculatedHeight = this.mirror_.getBoundingClientRect().height;
+  const calculatedWidth = this.mirror_.getBoundingClientRect().width;
+  // if the height is less then the window height just snap to the window height
+  if (calculatedHeight < this.initHeight_) {
+    this.container_.scrollTop = 0;
+    this.updateHeight(this.initHeight_);
+  } else {
+    this.updateHeight(calculatedHeight);
+  }
+
+  if (this.settings_.get('wraplines')) {
+    this.updateWidth(this.initWidth_);
+  } else {
+    this.updateWidth(Math.max(this.initWidth_, calculatedWidth));
+  }
+
+};
+
+
+EditorTextArea.prototype.updateLineNumbers = function() {
+  let linesRendered = this.lineNumbers_.children.length;
+
+  // remove excess elements
+  while (linesRendered > this.numLines_) {
+    this.lineNumbers_.lastChild.remove();
+    linesRendered--;
+  }
+  // add missing elements
+  while(linesRendered < this.numLines_) {
+    const line = linesRendered + 1;
+    // If word wrap is on we have to additionally set height
+    const height = this.settings_.get('wordwrap') ? this.getLineHeight(line) : undefined;
+    this.lineNumbers_.appendChild(this.lineNumberElem(line, height));
+    linesRendered++;
+  }
+
+};
+
+EditorTextArea.prototype.getLineHeight = function(lineNumber) {
+  // TODO(zafzal): Speed this up, splitting the whole doc is a bottleneck
+  const line = this.textarea_.value.split('\n')[lineNumber-1];
+  this.lineMirror_.innerText = line;
+  return this.lineMirror_.getBoundingClientRect().height;
+}
+
+EditorTextArea.prototype.undo = function() {
+  // This is handled by the text area defaults
+};
+
+EditorTextArea.prototype.redo = function() {
+  // This is handled by the text area defaults
+};
+
+EditorTextArea.prototype.focus = function() {
+  this.textarea_.focus();
+};
+
+/**
+ * @param {Session} session
+ * @param {string} extension
+ */
+EditorTextArea.prototype.setMode = function(session, extension) {
+  // textarea does not support any modes other then plain text :(
+};
+
+/**
+ * @param {number} fontSize
+ * Update font size from settings.
+ */
+EditorTextArea.prototype.setFontSize = function(fontSize) {
+  this.textarea_.style.fontSize = fontSize + 'px';
+  this.lineNumbers_.style.fontSize = fontSize + 'px';
+  this.mirror_.style.fontSize = fontSize + 'px';
+  this.lineMirror_.style.fontSize = fontSize + 'px';
+  this.updateLineNumbers();
+  this.updateTextArea();
+};
+
+/**
+ * @param {number} size
+ */
+EditorTextArea.prototype.setTabSize = function(size) {
```

Comment:
I think it's reasonable in a v0 to disable these settings in accessibility mode

Reply:
i believe it's actually a very easy css property you can set so no harm in keeping it

Reply:
Ack. For any settings not yet implemented I wouldn't block this on them.